Add global statistics postprocessor

gassmoeller opened the pull request:
This is a rebase from #1429. I will need to adjust the test outputs because of our switch to the 8.5 tester.

gassmoeller commented:
This should be ready for review/merge. For earlier comments that are addressed see #1429.

bangerth commented on source/simulator/solver.cc:
```diff
@@ -384,19 +384,39 @@ namespace aspect
+#if DEAL_II_VERSION_GTE(9,0,0)
+    SolverControl solver_control (1000, tolerance);
+#else
+    ExtendedSolverControl solver_control (1000, tolerance);
```
You no longer need the distinction here (i.e., just use `SolverControl` unconditionally) because for old versions of deal.II, writing `SolverControl` finds the declaration in namespace `aspect`, whereas for new versions of deal.II, it just takes the one in namespace `dealii` because there is none in the "closer namespace" `aspect`.

bangerth commented on include/aspect/compat.h:
```diff
@@ -31,4 +31,44 @@
+
+namespace dealii
+{
+  class ExtendedSolverControl : public SolverControl
```
just say
```
  namespace aspect {
    class SolverControl : public dealii::SolverControl
```
and you won't need the typedef below any more :-)

gassmoeller commented:
I addressed the comments, and it seems to work. Dont know why it did not work the first time I tried it that way. Anyway, should be ready to merge now.

tjhei commented:
This PR broke mainline:
```
In file included from /home/dealii/source/include/aspect/global.h:48:0,
                 from /home/dealii/source/source/boundary_composition/interface.cc:22:
/home/dealii/source/include/aspect/compat.h: In constructor â€˜aspect::SolverControl::SolverControl(unsigned int, double, bool, bool)â€™:
/home/dealii/source/include/aspect/compat.h:53:55: error: constructor delegates to itself
         SolverControl (n, tol, log_history, log_result)
```

bangerth commented:
On it.
